packages/monaco/src/browser/monaco-context-menu.ts

```diff
@@ -30,12 +30,10 @@ export class MonacoContextMenuService implements IContextMenuService {
 
     showContextMenu(delegate: IContextMenuDelegate): void {
         const anchor = toAnchor(delegate.getAnchor());
-        // If it is the general context menu, we want to delegate to our menu registry entirely and ignore the actually passed actions.
-        // Unfortunately checking the existence of certain properties seems to be the best way to tell, what kind of context menu is requested.
-        if (delegate.hasOwnProperty('getKeyBinding')) {
+        const actions = delegate.getActions();
+        if (actions.length > 0 && actions[0] instanceof monaco.actions.MenuItemAction) {
```

Comment:
@RomanNikitenko Could you add a comment why condition is like that? How actions look like for quick fixes?

Reply:
In case of editor context menu actions come as `MenuItemAction`
In case of `Quick Fix` actions come as `CodeActionAction`

Reply:
Is this way good enough? If so, I'll add the corresponding comment.

Reply:
yes, it is good